src/rocksdb.cc

```diff
@@ -170,9 +240,11 @@ rocksdb_open (rocksdb_t *db, rocksdb_open_t *req, const char *path, const rocksd
 
   strcpy(req->path, path);
 
+  req->mkdir_next = NULL;
+
   req->worker.data = static_cast<void *>(req);
 
-  return uv_queue_work(db->loop, &req->worker, rocksdb__on_open, rocksdb__on_status<rocksdb_open_t>);
+  return uv_queue_work(db->loop, &req->worker, rocksdb__recursive_mkdir, NULL);
```

Comment:
This still runs `uv_fs_mkdir()` from the thread pool, which isn't safe. It should run on the same thread that `rocksdb_open()` was called from.